### DISCUSSION TITLE
How I Can Adjust the input power

### DISCUSSION THREAD

**Discussion by salghanmi** [Telecominfraproject/oopt-gnpy#439] [Q&A]
Hi team,
I hope this message finds you well.

Firstly, I want to thank you for this great and useful tool.

I'm currently learning and evaluating the tool to simulate an optical path for optimization. 

The idea is to evaluate the SNR in the received end is lower than a threshold (the threshold can be entered by the user or manually specified), then increase the launch power to 1 mW and check again.

What I am doing this something like this: 
#initialisation
launch_power=0.5mW
SNR=read_SNR_from_Simulator()
#adjusting loop
while (SNR<SNR_target):
    launch_power=launch_power+1mW
print("New launch power: ", launch_power)
I don't know from where I can start, I read the Documntation but I can not figure it out. 
Where How Can I get the value of the SNR ?  do I need to work on the Docker or python 
Also,  Can you provide some exmples or toturials 
thank you

**Comment by jktjkt** [comment]
Hi @salghanmi, can you please clarify how exactly you're using GNPy? We don't have a `read_SNR_from_Simulator()` function in our code.

You can find some tutorials [in this older thread](https://github.com/Telecominfraproject/oopt-gnpy/discussions/378).
